### PR TITLE
Make loaders `list_data_models()`

### PR DESCRIPTION
**Pull request by uvsmtid-gs**
*   Define `data_model`-s by loaders.
*   Update GUI spinner to retain state between requests.
*   Optimize validation on server start (avoid scanning all `data_envelope` twice):
    *   Validate `index_prop`-s for all `data_envelope`.
    *   Validate that all `search_prop`-s are listed as `index_prop`-s. 
*   Spec `FS_45_08_22_15.data_model_manipulation.md`.
*   Spec `TODO_89_50_17_63.fine_tune_list_of_index_props.md`.
*   Define `ReservedEnvelopeClass.ClassNoData`.
*   Rename `LoadProgressState` to `ProgressTracker`.

### CODE REVIEW

File: src/argrelay/_version.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # See `docs/dev_notes/version_format.md`:
 # Implements this:
 # https://stackoverflow.com/a/7071358/441652
-__version__ = "0.7.19"
+__version__ = "0.7.20.dev0"
```
Comment (uvsmtid) [inline src/argrelay/_version.py:4]: Finalize before merge.

File: docs/feature_stories/FS_45_08_22_15.data_model_manipulation.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+feature_story: FS_45_08_22_15
+feature_title: data model manipulation
+feature_status: PARTIAL
+---
+
+TODO: Move details about metadata from FS_74_69_61_79 get set data envelope here.
```
Comment (uvsmtid) [inline docs/feature_stories/FS_45_08_22_15.data_model_manipulation.md:7]: Do before merge.

File: docs/feature_stories/FS_74_69_61_79.get_set_data_envelope.md [hunk 3]
```diff
@@ -5,6 +5,9 @@ feature_status: TODO
 ---
 
 TODO: Maybe rename "get set data envelope" to "get set envelope_container"?
+      Actually, it is all about selecting a snapshot of `data_envelope`-s - maybe "get set snapshot"?
+
+TODO: Split metadata details into FS_45_08_22_15 data model manipulation.
```
Comment (uvsmtid) [inline docs/feature_stories/FS_74_69_61_79.get_set_data_envelope.md:10]: Do before merge.

File: docs/task_refs/TODO_89_50_17_63.fine_tune_list_of_index_props.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+TODO: TODO_89_50_17_63: fine-tune list of `index_props`
```
Comment (uvsmtid) [inline docs/task_refs/TODO_89_50_17_63.fine_tune_list_of_index_props.md:2]: Do for `GitRepoLoader`.

File: tests/offline_tests/config_only_plugins/test_config_only_plugins.py [hunk 5]
```diff
@@ -232,11 +234,19 @@ def test_config_only_invocation(self):
                     expected_exit_code = next(iter(popen_mock_config.values()))[0]
                     self.assertEqual(cm.exception.code, expected_exit_code)
 
+    # TODO: Unskip:
+    @unittest.skip
```
Comment (uvsmtid) [inline tests/offline_tests/config_only_plugins/test_config_only_plugins.py:238]: Unskip before merge.

File: docs/task_refs/TODO_89_50_17_63.fine_tune_list_of_index_props.md (hunk 4 above)
Comment (uvsmtid) [reply docs/task_refs/TODO_89_50_17_63.fine_tune_list_of_index_props.md:2]: Ignore.

File: src/argrelay/_version.py (hunk 1 above)
Comment (uvsmtid-gs) [reply src/argrelay/_version.py]: Done.

File: docs/task_refs/TODO_89_50_17_63.fine_tune_list_of_index_props.md (hunk 4 above)
Comment (uvsmtid-gs) [reply docs/task_refs/TODO_89_50_17_63.fine_tune_list_of_index_props.md:2]: Next time.

File: tests/offline_tests/config_only_plugins/test_config_only_plugins.py (hunk 5 above)
Comment (uvsmtid-gs) [reply tests/offline_tests/config_only_plugins/test_config_only_plugins.py]: Done.

File: docs/feature_stories/FS_45_08_22_15.data_model_manipulation.md (hunk 2 above)
Comment (uvsmtid-gs) [reply docs/feature_stories/FS_45_08_22_15.data_model_manipulation.md:7]: Next time.

File: docs/feature_stories/FS_74_69_61_79.get_set_data_envelope.md (hunk 3 above)
Comment (uvsmtid-gs) [reply docs/feature_stories/FS_74_69_61_79.get_set_data_envelope.md:10]: Next time.

**Review by uvsmtid — DISMISSED** [review]

**Review by uvsmtid — APPROVED** [review]

**Pull request merged**